src/unit.rs

```diff
@@ -792,27 +792,31 @@ impl<'input, 'abbrev, 'unit, Endian> DebuggingInformationEntry<'input, 'abbrev,
 
     /// Find the first attribute in this entry which has the given name,
     /// and return it. Returns `Ok(None)` if no attribute is found.
-    pub fn attr(&self, name: constants::DwAt) -> Option<Attribute<'input, Endian>> {
+    pub fn attr(&self, name: constants::DwAt) -> Result<Option<Attribute<'input, Endian>>> {
         let mut attrs = self.attrs();
-        while let Ok(Some(attr)) = attrs.next() {
+        while let Some(attr) = try!(attrs.next()) {
             if attr.name() == name {
-                return Some(attr);
+                return Ok(Some(attr));
             }
         }
-        None
+        Ok(None)
     }
 
     /// Find the first attribute in this entry which has the given name,
     /// and return its raw value. Returns `Ok(None)` if no attribute is found.
-    pub fn attr_value_raw(&self, name: constants::DwAt) -> Option<AttributeValue<'input, Endian>> {
-        self.attr(name).map(|attr| attr.raw_value())
+    pub fn attr_value_raw(&self,
+                          name: constants::DwAt)
+                          -> Result<Option<AttributeValue<'input, Endian>>> {
+        self.attr(name).map(|attr| attr.map(|attr| attr.raw_value()))
```

Comment:
An alternative is `Ok(self.attr(name)?.map(|attr| attr.raw_value())`. Not sure if that is better though.